### PR TITLE
Mx 15523 new abi update

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- New ABI-go version integration
- Refactor for some sovereign tests that use abi serializer
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/chainSimulator/tests/bridge/common.go [hunk 1]
```diff
@@ -61,3 +69,70 @@ func DeploySovereignBridgeSetup(
...
+			"@" + hex.EncodeToString([]byte(token.Identifier)) +
+			"@" + getNonceArg(t, dtaCodec, token.Nonce) +
+			"@" + hex.EncodeToString(token.Amount.Bytes())
+	}
+
+	transferDataArg := getTransferDataArg(t, dtaCodec, transferData)
+	depositArgs = depositArgs +
+		"@" + hex.EncodeToString([]byte(deposit)) +
+		"@" + hex.EncodeToString(receiver) +
+		transferDataArg
```
Comment (andreibancioiu) [inline cmd/sovereignnode/chainSimulator/tests/bridge/common.go:103]: Generally speaking, it's possible to do this using the serializer, as well. Can also stay as it is.

**Review by andreibancioiu — DISMISSED** [review]

File: cmd/sovereignnode/chainSimulator/tests/bridge/common.go (hunk 1 above)
Comment (axenteoctavian) [reply cmd/sovereignnode/chainSimulator/tests/bridge/common.go]: This sounds good. I will refactor this.

**Review by andreibancioiu — DISMISSED** [review]

File: cmd/sovereignnode/dataCodec/interface.go [hunk 2]
```diff
@@ -29,6 +25,7 @@ type OperationDataEncoder interface {
 
 // SovereignDataCodec is the interface for serializing/deserializing data
 type SovereignDataCodec interface {
+	abiSerializer.AbiSerializer
```
Comment (mariusmihaic) [inline cmd/sovereignnode/dataCodec/interface.go:28]: This sovereign data codec should only have the responsibility of working with our custom structs to serialize/deserialize.

It should only receive the `Serializer` interface, not embed it and extend it.

Also, you added it here just for testing purposes, which is not ok.

Please revert these changes.

If you really need the serializer, you should have in runTypeComps

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]
